api/src/main/java/com/xwiki/googleapps/internal/CookieAuthenticationPersistenceImpl.java

```diff
@@ -27,42 +27,47 @@
 import javax.crypto.NoSuchPaddingException;
 import javax.crypto.spec.SecretKeySpec;
 import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
 import javax.servlet.http.Cookie;
 
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.lang3.StringUtils;
-
 import org.slf4j.Logger;
-import org.xwiki.apps.googleapps.CookieAuthenticationPersistence;
+import com.xwiki.googleapps.GoogleAppsManager;
 import org.xwiki.component.annotation.Component;
 import org.xwiki.component.annotation.InstantiationStrategy;
 import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
-import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.configuration.ConfigurationSource;
 import org.xwiki.stability.Unstable;
 
 import com.xpn.xwiki.XWikiContext;
-import com.xpn.xwiki.XWikiException;
 
 /**
- * Tools to help storing and retrieving enriched information within cookies such as the
- * linked Google user profile.
- *
- * Copied code from xwiki-authenticator-trusted
+ * Tools to help storing and retrieving enriched information within cookies such as the linked Google user profile.
+ * <p>
+ * This code is inspired by  from xwiki-authenticator-trusted
  * https://github.com/xwiki-contrib/xwiki-authenticator-trusted/edit/master\
```

Comment:
I would put a link around ``xwiki-authenticator-trusted`` and I would use a tag instead of the master branch, which is not stable.